pydicom/filewriter.py

```diff
@@ -1024,33 +1024,41 @@ def dcmwrite(
         ``save_as()`` wraps ``dcmwrite()``.
     """
     tsyntax: Optional[UID]
+    try:
+        tsyntax = dataset.file_meta.TransferSyntaxUID
+    except AttributeError:
+        tsyntax = None
+
+    cls_name = dataset.__class__.__name__
+    encoding = (dataset.is_implicit_VR, dataset.is_little_endian)
 
     # Ensure is_little_endian and is_implicit_VR are set
-    if None in (dataset.is_little_endian, dataset.is_implicit_VR):
-        has_tsyntax = False
-        try:
-            tsyntax = dataset.file_meta.TransferSyntaxUID
-            if not tsyntax.is_private:
-                dataset.is_little_endian = tsyntax.is_little_endian
-                dataset.is_implicit_VR = tsyntax.is_implicit_VR
-                has_tsyntax = True
-        except AttributeError:
-            pass
+    if None in encoding:
+        if tsyntax and not tsyntax.is_private:
+            dataset.is_little_endian = tsyntax.is_little_endian
+            dataset.is_implicit_VR = tsyntax.is_implicit_VR
 
-        if not has_tsyntax:
-            name = dataset.__class__.__name__
+        if tsyntax is None:
```

Comment:
Nitpicking: could move this before the other check and avoid checking tsyntax again

Reply:
Done